### PR TITLE
(DO NOT MERGE)(PUP-4088) Fix new path acceptance for some Windows

### PR DESCRIPTION
**Pull request by MikaelSmith**
Without this change, the new ensure_puppet-agent_paths acceptance test
fails on Windows Server 2003, and when testing 32-bit puppet-agent
installs on 64-bit Windows.

Update the test to use the correct paths for both cases.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Running through acceptance tests for this.

File: acceptance/tests/ensure_puppet-agent_paths.rb [hunk 1]
```diff
@@ -90,17 +114,37 @@
...
+    return '/usr/bin'
+  end
+
+  case platform
+  when /windows/
+    # If undefined, ruby_arch defaults to x86
+    if ruby_arch == 'x64'
+      ruby_arch = /-64/
+    else
+      ruby_arch = /-32/
```
Comment (joshcooper) [inline acceptance/tests/ensure_puppet-agent_paths.rb]: does `-` need to be escaped?

Comment (MikaelSmith) [inline acceptance/tests/ensure_puppet-agent_paths.rb]: Quite probably... I'm surprised that hasn't caused a test issue.

Comment (MikaelSmith) [inline acceptance/tests/ensure_puppet-agent_paths.rb]: It's valid, quick test: `'windows-2003r2-64' =~ /-2003/`.

It's only a keyword in a square-bracket expression, such as /[2-3]/.

**Comment by MikaelSmith** [comment]
Tested with MSI from SHA 215e36f962dfd7080947d78a1a1340b0aadcfa62, using x86 build on Windows Server 2003 (x86/x64), and x86 and x64 builds on Windows Server 2012 x64.

Also tested with git on all those platforms; a minor fix is needed to run with git that looks unrelated, so I'll post it as a separate PR.

**Comment by MikaelSmith** [comment]
This should be ready to go, passed all my testing.

File: acceptance/tests/ensure_puppet-agent_paths.rb [hunk 2]
```diff
@@ -6,53 +6,78 @@
...
-]
+def config_options(platform)
+  case platform
+  when /windows/
+    if platform =~ /2003/
+      common_app_data = 'C:/Documents and Settings/All Users/Application Data'
+    else
+      common_app_data = 'C:/ProgramData'
+    end
+    puppet_install = "#{common_app_data}/PuppetLabs"
```
Comment (Iristyle) [inline acceptance/tests/ensure_puppet-agent_paths.rb]: Minor, but this really `puppet_data`, not `puppet_install`

File: acceptance/tests/ensure_puppet-agent_paths.rb (hunk 1 above)
Comment (Iristyle) [inline acceptance/tests/ensure_puppet-agent_paths.rb]: Right, `-` should only be meaningful in character classes...

**Comment by Iristyle** [comment]
Now that I'm not on the VPN (over a DSL connection), I'm going to run this through acceptance locally to make sure we're good.

Shouldn't take too long to verify / merge.

**Comment by Iristyle** [comment]
Note that the test is running very slowly querying for settings individually.  @MikaelSmith is changing to `puppet agent --configprint all`

**Comment by MikaelSmith** [comment]
Pushed a change to do configprint all.

**Pull request merged**